### PR TITLE
feat(skore): Support passing `sklearn` metrics without "neg_" 

### LINKED ISSUE #2607: feat: Make it possible to pass sklearn metrics without `"neg_"`
### Is your feature request related to a problem? Please describe.

The following works:

```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier
from skore import evaluate

X, y = make_classification(random_state=0)
report = evaluate(
    RandomForestClassifier(max_depth=2, random_state=0), X, y, pos_label=1
)
report.metrics.summarize(metric='neg_mean_squared_error').frame()
```

But this doesn't:
```python
report.metrics.summarize(metric='mean_squared_error').frame()
```

### Describe the solution you'd like

Make it work

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by direkkakkar319-ops**
#### Reference issue 
Fixes #2607 

#### Change description
1. Feature: Allow passing scikit-learn metrics without the neg_ prefix
2. Key Changes:
* Added a `_METRIC_ALIASES` dictionary and a `_resolve_metric_alias` fallback in the _`MetricsAccessor ` .
* Added a dedicated and comprehensive test file (`test_metric_aliases.py`)
* Updated the displays user guide documentation in Sphinx
3. Compatibility / Breaking Changes: No breaking changes. 
5 .testing
testing 
```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier
from skore import evaluate

X, y = make_classification(random_state=0)
report = evaluate(
    RandomForestClassifier(max_depth=2, random_state=0), X, y, pos_label=1
)

#This used to throw a ValueError!
print(report.metrics.summarize(metric='mean_squared_error').frame())
```

output:
```python
RandomForestClassifier
Metric                                    
Mean Squared Error                    0.05
```

#### Contribution checklist
- [x] Unit tests were added or updated
- [x] Documentation was added or updated
- [x] The code passes our style conventions
- [x] All the tests pass
- [x] The documentation builds and renders properly
- [x] A new changelog entry was added to CHANGELOG.rst 
- [x] All the commits in the PR are signed
- [x] The pull request title respects the Conventional Commits convention

#### AI usage disclosure
AI tools were involved for:
- [ ] Code generation
- [x] Test generation
- [ ] Documentation
- [ ] Research and understanding

pytest pass locally
command used: `cd skore && python -m pytest`
Screenshot 2026-04-09 150235

pre-commits pass locally
command used: `pre-commit run --all-files`
Screenshot 2026-04-09 151334

sphinx test pass locally
command used: `cd sphinx && make html`
image

pytest for tests i added 
command used: ` pytest skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py`
image

### CODE REVIEW

**Review by auguste-probabl — CHANGES_REQUESTED** [review]
I expected that logic to go in `metrics.py`, near where we run `sklearn.metrics.get_scorer`; that way the whole logic is there.

File: skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+@pytest.mark.parametrize("metric", ["accuracy", "f1", "r2", "neg_mean_squared_error"])
+def test_resolve_metric_alias_passthrough(metric):
+    """Non-aliased strings (including neg_ prefixed ones) pass through unchanged."""
+    assert _MetricsAccessor._resolve_metric_alias(metric) == metric
+
+
+def test_resolve_metric_alias_non_string():
+    """Non-string metrics (e.g. callables) are returned unchanged."""
+    func = lambda y_true, y_pred: 0.0  # noqa: E731
+    assert _MetricsAccessor._resolve_metric_alias(func) is func
```
Comment (auguste-probabl) [inline skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py]: The functionality will be tested at the summarize level so no need for these tests

File: skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py
Comment (auguste-probabl) [inline skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py:1]: Don't create a whole file for these tests, add them to the `summarize` tests

File: sphinx/user_guide/reporters.rst
Comment (auguste-probabl) [inline sphinx/user_guide/reporters.rst:1]: This functionality should get a changelog entry :)

**Comment by direkkakkar319-ops** [comment]
after the changes:
1. pytest pass 
image

2. pre-commits pass 
image

File: skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py (hunk 1 above)
Comment (direkkakkar319-ops) [reply skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py]: removed the tests that would be tested at summarise level

File: skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py
Comment (direkkakkar319-ops) [reply skore/tests/unit/reports/estimator/metrics/test_metric_aliases.py:1]: added tests to test_summarize.py and deleted the new created test file

File: sphinx/user_guide/reporters.rst
Comment (direkkakkar319-ops) [reply sphinx/user_guide/reporters.rst:1]: added a changelog entry for the function

**Comment by direkkakkar319-ops** [comment]
Just ran demo.py(fil i created locally for testing) via WSL
```python
Generating data and training model...
--- Testing without 'neg_' prefix ---
report.metrics.summarize(metric='mean_squared_error').frame()
                    RandomForestClassifier
Metric                                    
Mean Squared Error                    0.05
--- Testing with 'neg_' prefix (for comparison) ---
report.metrics.summarize(metric='neg_mean_squared_error').frame()
                    RandomForestClassifier
Metric                                    
Mean Squared Error                    0.05
```

**Comment by direkkakkar319-ops** [comment]
dem.py for reviewing 
```python

file code for checkoing 
```python
from sklearn.datasets import make_classification
from sklearn.ensemble import RandomForestClassifier
from skore import evaluate

print("Generating data and training model...")
X, y = make_classification(random_state=0)
report = evaluate(
    RandomForestClassifier(max_depth=2, random_state=0), X, y, pos_label=1
)

print("\n--- Testing without 'neg_' prefix ---")
print("report.metrics.summarize(metric='mean_squared_error').frame()")
print(report.metrics.summarize(metric='mean_squared_error').frame())

print("\n--- Testing with 'neg_' prefix (for comparison) ---")
print("report.metrics.summarize(metric='neg_mean_squared_error').frame()")
print(report.metrics.summarize(metric='neg_mean_squared_error').frame())
```

**Comment by auguste-probabl** [comment]
@direkkakkar319-ops Thanks for the quick update! Btw, there is no need to provide screenshots of passing checks; the CI will tell us reviewers if your changes pass the bar. Same for the demo scripts: the unit tests will show us that the feature works as intended, and the documentation will explain how the feature works.

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 2]
```diff
@@ -74,7 +74,7 @@ def _parse_metrics(
         if metric is None or (isinstance(metric, Sized) and len(metric) == 0):
             items = [(None, m) for m in self._parent._metric_registry]
         elif isinstance(metric, dict):
-            items = list(metric.items())
+            items = [(name, m) for name, m in metric.items()]
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: This doesn't change the logic so I'd rather not change that line

File: skore/src/skore/_sklearn/metrics.py [hunk 3]
```diff
@@ -28,6 +28,34 @@ def _select_kwargs(func: Callable, kwargs: dict[str, Any]) -> dict[str, Any]:
...
+    "root_mean_squared_error": "neg_root_mean_squared_error",
+    "root_mean_squared_log_error": "neg_root_mean_squared_log_error",
+    "mean_poisson_deviance": "neg_mean_poisson_deviance",
+    "mean_gamma_deviance": "neg_mean_gamma_deviance",
+    "max_error": "neg_max_error",
+    "negative_likelihood_ratio": "neg_negative_likelihood_ratio",
+}
+
+
+def _resolve_metric_alias(metric: str) -> str:
```
Comment (auguste-probabl) [inline skore/src/skore/_sklearn/metrics.py]: No need to create a whole function for one line of logic

File: skore/tests/unit/reports/estimator/metrics/test_summarize.py [hunk 4]
```diff
@@ -1070,3 +1070,27 @@ def test_data_source_both(forest_binary_classification_data):
...
+
+
+@pytest.mark.parametrize(
+    "metric_without_neg, metric_with_neg",
+    [
+        ("mean_squared_error", "neg_mean_squared_error"),
+        ("mean_absolute_error", "neg_mean_absolute_error"),
+        ("root_mean_squared_error", "neg_root_mean_squared_error"),
+    ],
+)
```
Comment (auguste-probabl) [inline skore/tests/unit/reports/estimator/metrics/test_summarize.py]: No need to have 3 tests for these, just choose one example and use that

File: sphinx/user_guide/reporters.rst [hunk 5]
```diff
@@ -84,7 +84,9 @@ regression). Nevertheless, you can specify the metrics you want to compute thank
 `scoring` parameter. We accept different types:
```
Comment (auguste-probabl) [inline sphinx/user_guide/reporters.rst]:
```suggestion
`metric` parameter. We accept different types:
```

Comment (auguste-probabl) [reply sphinx/user_guide/reporters.rst]: while we're at it

File: CHANGELOG.rst
Comment (auguste-probabl) [inline CHANGELOG.rst:1]: The change needs to be in `Unreleased` section; the other sections are for previously released versions of skore so they cannot be edited.

**Review by auguste-probabl — CHANGES_REQUESTED** [review]
Good progress, almost there

File: CHANGELOG.rst
Comment (direkkakkar319-ops) [reply CHANGELOG.rst:1]: changes are only on unreleased section now

File: sphinx/user_guide/reporters.rst (hunk 5 above)
Comment (direkkakkar319-ops) [reply sphinx/user_guide/reporters.rst]: done

File: skore/tests/unit/reports/estimator/metrics/test_summarize.py (hunk 4 above)
Comment (direkkakkar319-ops) [reply skore/tests/unit/reports/estimator/metrics/test_summarize.py]: now we use one example only that is `"mean_squared_error"`

File: skore/src/skore/_sklearn/metrics.py (hunk 3 above)
Comment (direkkakkar319-ops) [reply skore/src/skore/_sklearn/metrics.py]: removed the function and now we will use .get()

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py (hunk 2 above)
Comment (direkkakkar319-ops) [reply skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: made the line same

**Review by auguste-probabl — DISMISSED** [review]

File: CHANGELOG.rst [hunk 6]
```diff
@@ -50,6 +50,12 @@ Changed
 Added
 -----
 
+- :meth:`~EstimatorReport.metrics.summarize` (and equivalent methods on
+  :class:`~skore.CrossValidationReport` and :class:`~skore.ComparisonReport`) now
+  accept scikit-learn metric names without the `neg_` prefix. For example,
+  ``"mean_squared_error"`` can be passed instead of ``"neg_mean_squared_error"``
+  and is resolved automatically.
+
```
Comment (auguste-probabl) [inline CHANGELOG.rst]:
```suggestion
  ``"mean_squared_error"`` can be passed instead of ``"neg_mean_squared_error"``
  and is resolved automatically. See :pr:`2735` by :user:`direkkakkar319-ops`.
```

**Review by auguste-probabl — APPROVED** [review]
Thank you!!

**Comment by direkkakkar319-ops** [comment]
thanks for the merge @auguste-probabl : )
